src/main.ts

```diff
@@ -1,8 +1,30 @@
 import { NestFactory } from '@nestjs/core';
 import { AppModule } from './app.module';
+import { LoggerService } from '@tazama-lf/frms-coe-lib';
+import { validateProcessorConfig } from '@tazama-lf/frms-coe-lib/lib/config';
+import { ValidationPipe } from '@nestjs/common';
 
 async function bootstrap() {
-  const app = await NestFactory.create(AppModule);
-  await app.listen(process.env.PORT ?? 3000);
+  // Validate and get the processor config for LoggerService
+  const configuration = validateProcessorConfig();
+  const logger = new LoggerService(configuration);
+
+  const app = await NestFactory.create(AppModule, {
+    logger,
+  });
```

Comment:
NestJS uses a ConfigModule to load environment variables. Here you are passing the loggerService some configuration even before the NestFactory is initialized. Therefore, when I was trying to run it the loggerService seemed to throw and undefined env error. 
```suggestion
  const app = await NestFactory.create(AppModule);
  const logger = app.get(LoggerService);
  app.useLogger(logger);
```

Reply:
We have updated our NestJS bootstrap process to follow best practices for environment variable loading and logger initialization. The application now initializes the NestFactory first, retrieves the LoggerService from the DI container, and sets it using app.useLogger(logger). This ensures all configuration is loaded before the logger is used, resolving the previous undefined environment error.